xmrstak/backend/amd/amd_gpu/gpu.cpp

```diff
@@ -522,7 +522,13 @@ std::vector<GpuContext> getAMDDevices(int index)
 
 			// if environment variable GPU_SINGLE_ALLOC_PERCENT is not set we can not allocate the full memory
 			ctx.deviceIdx = k;
-			ctx.freeMem = std::min(ctx.freeMem, maxMem);
+			/*
+			* System does __not always__ report true memory left, 
+			* only the max amount we can allocate at once...
+			* It is a __hack__ to not be querying the max alloc before each
+			* and every alloc if trying to figure out the amount available.
+			*/
+			ctx.freeMem = maxMem * std::floor(ctx.freeMem / maxMem);
```

Comment:
I found out why Vega users were reporting less than 4GB being reported in amd.txt.

https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clGetDeviceInfo.html
CL_DEVICE_MAX_MEM_ALLOC_SIZE
Max size of memory object allocation in bytes. The minimum value is max (1/4th of CL_DEVICE_GLOBAL_MEM_SIZE, 128*1024*1024)

The gotcha is that the allocation size reported does not have to be the max mem size. This commit here is a HACK and needs to be replaced with something different. The proper code might need to query the devices at runtime than ahead of time!